.github/workflows/vib-publish.yaml

```diff
@@ -0,0 +1,113 @@
+jobs:
+  get-chart:
+    runs-on: ubuntu-latest
+    name: 'Get modified charts'
+    outputs:
+      chart: ${{ steps.get-chart.outputs.chart }}
+      result: ${{ steps.get-chart.outputs.result }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: charts
+          fetch-depth: 2 # to be able to obtain files changed in the latest commit
+      - id: get-chart
+        name: 'Get modified charts'
+        run: |
+          cd charts
+          files_changed="$(git show --pretty="" --name-only)"
+          # Adding || true to avoid "Process exited with code 1" errors
+          charts_dirs_changed="$(echo "$files_changed" | xargs dirname | grep -o "bitnami/[^/]*" | sort | uniq || true)"
+          # Using grep -c as a better alternative to wc -l when dealing with empty strings."
+          num_charts_changed="$(echo "$charts_dirs_changed" | grep -c "bitnami" || true)"
+          num_version_bumps="$(echo "$files_changed" | grep Chart.yaml | xargs git show | grep -c "+version" || true)"
+
+          if [[ "$num_charts_changed" -ne "$num_version_bumps" ]]; then
+            # Changes done in charts but version not bumped -> ERROR
+            echo "::set-output name=error::Detected changes in charts without version bump in Chart.yaml.\nCharts changed: ${num_charts_changed}\n${charts_dirs_changed}\nVersion bumps detected: ${num_version_bumps}"
+            echo "::set-output name=result::fail"
+          elif [[ "$num_charts_changed" -eq "1" ]]; then
+            # Changes done in only one chart -> OK
+            chart_name=$(echo "$charts_dirs_changed" | sed "s|bitnami/||g")
+            echo "::set-output name=chart::${chart_name}"
+            echo "::set-output name=result::ok"
+          else
+            # Changes done in more than chart -> FAIL
+            echo -e "::set-output name=error::Changes detected in more than one chart directory:\n${charts_dirs_changed}\nThe publish process will be stopped. Please create different commits for each chart."
+            echo "::set-output name=result::fail"
+          fi
+      - id: show-error
+        name: 'Show error'
+        if: ${{ steps.get-chart.outputs.result == 'fail' }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+              core.setFailed('${{ steps.get-chart.outputs.error }}')
+
+  vib-publish:
+    runs-on: ubuntu-latest
+    needs: get-chart
+    if: ${{ needs.get-chart.outputs.result == 'ok' }}
+    environment: vmware-image-builder
+    name: Publish
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout Repository
+        with:
+          path: charts
+      - uses: vmware-labs/vmware-image-builder-action@main
+        name: Verify and publish ${{ needs.get-chart.outputs.chart }}
+        with:
+          pipeline: ${{ needs.get-chart.outputs.chart }}/vib-publish.json
+          config: charts/.vib/
+        env:
+          TARGET_PLATFORM: 91d398a2-25c4-4cda-8732-75a3cfc179a1
```

Comment:
```suggestion
          VIB_ENV_TARGET_PLATFORM: 91d398a2-25c4-4cda-8732-75a3cfc179a1
```

Reply:
Additionally, I think we should be using the same approach we follow for `vib-verify` now:

1. Do not hardcode the value for `target-platform` here and use a secret instead. That way we prevent adding commits just to simply update the platform.
2. Use TKG by default and add another variable for assets incompatible with this platform

https://github.com/bitnami/charts/blob/5571b65f6b54f74097792c161a47566f755cc01d/.github/workflows/vib-verify.yaml#L93-L96

Reply:
> That way we prevent adding commits just to simply update the platform.

The downside of using secrets is that it's not easy to retrieve the content of the secret once it's defined. Also, we could argue if that's actually a pro and not a con 😄 (configuration as code)
Having a way of tracing why the target-platform and follow the standard code review process is a feature in my opinion.

Reply:
> Use TKG by default and add another variable for assets incompatible with this platform
We cannot follow this approach, since we can only launch 1 pipeline (otherwise the publish phase would be executed twice). In addition, we would also verifying twice something that was already checked during PR verify.

Reply:
> Additionally, I think we should be using the same approach we follow for `vib-verify` now:
> 
>     2. Use TKG by default and add another variable for assets incompatible with this platform
> 
> 
> https://github.com/bitnami/charts/blob/5571b65f6b54f74097792c161a47566f755cc01d/.github/workflows/vib-verify.yaml#L93-L96

@jsalmeron I think you didn't cover this.